python/ray/tests/test_object_spilling.py

```diff
@@ -278,34 +314,42 @@ def depends(arg):
 def test_spill_objects_automatically(object_spilling_config, shutdown_only):
     # Limit our object store to 75 MiB of memory.
     ray.init(
+        num_cpus=1,
         object_store_memory=75 * 1024 * 1024,
         _system_config={
-            "max_io_workers": 4,
+            "max_io_workers": 8,
             "automatic_object_spilling_enabled": True,
             "object_store_full_max_retries": 4,
             "object_store_full_initial_delay_ms": 100,
             "object_spilling_config": object_spilling_config,
+            "min_spilling_size": 30 * 1024 * 1024,
         })
-    arr = np.random.rand(1024 * 1024)  # 8 MB data
     replay_buffer = []
+    solution_buffer = []
+    buffer_length = 100
 
     # Wait raylet for starting an IO worker.
     time.sleep(1)
 
     # Create objects of more than 800 MiB.
-    for _ in range(100):
+    for _ in range(buffer_length):
         ref = None
         while ref is None:
+            multiplier = random.choice([1, 2, 3])
+            arr = np.random.rand(multiplier * 1024 * 1024)
             ref = ray.put(arr)
             replay_buffer.append(ref)
+            solution_buffer.append(arr)
 
     print("-----------------------------------")
 
     # randomly sample objects
     for _ in range(1000):
-        ref = random.choice(replay_buffer)
+        index = random.choice([i for i in range(buffer_length)])
+        ref = replay_buffer[index]
+        solution = solution_buffer[index]
         sample = ray.get(ref, timeout=0)
-        assert np.array_equal(sample, arr)
+        assert np.array_equal(sample, solution)
```

Comment:
Added to make sure we are reading the correct offsets. (Without this, although we don't read correct offsets, this can succeed. For example, imagine we are always reading from offset 0. This will always succeed in the previous test case).